src/main/java/com/google/sps/data/Organization.java

```diff
@@ -47,4 +52,89 @@ public Organization(Entity entity) {
      this.urlLink = (String) entity.getProperty("orgUrl");
      this.moderators = (ArrayList) entity.getProperty("moderatorList");
   }
+
+  public void updateOrganization(HttpServletRequest request, boolean isMaintainer, boolean isModerator) throws IllegalArgumentException{
+    Map<String, String> properties = new HashMap<String,String>();
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String formKey = entry.getKey();
+      String propertyKey = entry.getValue();
+      String formValue = "";
+
+      // will only get approval param is a maintainer sent the request
+      if(propertyKey.equals("isApproved") && !isMaintainer) {
+        continue;
+      }
+
+      // will only get moderatorList param if request is sent by a moderator or maintainer
+      if(propertyKey.equals("moderatorList") && !isModerator && !isMaintainer) {
+        continue;
+      }
+
+      // setting organization description is optional
```

Comment:
Why? Won't this clear out the organization description?

Reply:
added why we are doing this in the design doc